### ISSUE TITLE
add an optional "readOnly" property for items

### ISSUE THREAD

**Issue by mamund** [collection-json/spec#12]
add an _optional_ `readOnly` property for items. This will make it easier for clients to know which items are eligible for update or deletion. 

`readOnly="true"` : this item SHOULD NOT be edited using the supplied template and is not delete-able
`readOnly="false"` : this item MAY be edited using the supplied template 
if the `readOnly` property is missing that item SHOULD be assumed to be eligible for edits and deletions.

Document authors can still include entries in the `links` collection for each item that point to representations that include a template for use in editing (e.g. `rel="edit"` or `rel="delete"`). This will make it easier to indicate custom edit templates for individual items.

_This is a backward-compatible change. There is no breaking change to the format. Clients SHOULD already ignore properties they do not know._

**Comment by carlesjove** [comment]
There's a [registered extension](https://github.com/collection-json/extensions/blob/master/read-only.md) for this. I think the only thing to clarify here would be the name: `camelCase` vs `separate-words`. At the registered extension they use `separate-words`.

**Comment by mamund** [comment]
yep, this is an attempt to move the extension into the main spec doc.  i've found this extension very useful!

so, onto the naming. i _prefer_ not to introduce dashes into Cj since it adds some burden to parsing the JSON since "dashed" properties need to be addressed using brackets:

`items[0]["read-only"]` vs. `items[0].readOnly`

i am also ok w/ using all lower case:

`items[0].readonly`

finally, would love to hear from others using the extension. how do you feel about adding this to the spec? does changing the name cause enough problems to make a case for just leaving it as a "dashed" element? do you find the dash causing any added burden in parsing (as i claim)? or is it no big deal?

**Comment by carlesjove** [comment]
Hi @mamund 

I'm already [including extensions](https://github.com/carlesjove/collection_json_serializer/tree/dev#extensions) on the Ruby CJ::Serializer. For now, I'll add "read only" as an extension, and will later move it to the spec if it's finally merged.

Just to clarify:
1. The attribute name is finally `readOnly`, right?
2. Is it ok if I send a PR to [the extension draft](https://github.com/collection-json/extensions/blob/master/read-only.md), so we have a naming consistency all around?

cheers

**Comment by mamund** [comment]
well, we can't change the existing extension since others maybe depending
on it already.

i've not finally decided on whether to change to readonly or readOnly and
was hoping to hear from the author of the existing extension, too.

for now, go ahead and do a PR for a new extension and we can sort this out
shortly.

cool?

**Comment by carlesjove** [comment]
I'll rather wait until you decide on the name. It's not really an urgency.

Let's ping @pmhsfelix and see what he's got to say ;-)
